packages/autocomplete-plugin-algolia-insights/src/createAlgoliaInsightsPlugin.ts

```diff
@@ -84,11 +87,41 @@ export function createAlgoliaInsightsPlugin(
   options: CreateAlgoliaInsightsPluginParams
 ): AutocompletePlugin<any, undefined> {
   const {
-    insightsClient,
+    insightsClient: providedInsightsClient,
     onItemsChange,
     onSelect: onSelectEvent,
     onActive: onActiveEvent,
   } = getOptions(options);
+  let insightsClient: InsightsClient = providedInsightsClient || noop;
```

Comment:
```suggestion
  let insightsClient: InsightsClient = providedInsightsClient || noop;
```

are we ever in the noop branch?

Reply:
The client is now optional, so `providedInsightsClient` could be `undefined`.

If the logic we have within `safelyRunOnBrowser` fails to assign a proper client to the local `insightsClient` (not present in the page, fails to load from jsDelivr) _or_ is assigned after there's a call to send an event, we'll get an error "insightsClient is not a function".

Reply:
that's only the case if the code right underneath is somehow reverted, in which case we would still not be in the noop branch.

I'm ok with being extra safe for not needlessly throwing, but I don't think the noop ever is used.

Reply:
Ah you're right, there's the queue assignment. TypeScript isn't able to infer it properly though so I can just cast it, and we can rely on tests.

See [66e45f0](https://github.com/algolia/autocomplete/pull/1106/commits/66e45f0f8f494a6712ffb2059a75c3164a7e440d)